app/src/hooks/api/useSurveyApi.ts

```diff
@@ -110,6 +110,38 @@ const useSurveyApi = (axios: AxiosInstance) => {
    * @return {*}  {Promise<string[]>}
    */
   const uploadSurveyAttachments = async (
+    projectId: number,
+    surveyId: number,
+    file: File,
+    attachmentType?: string,
+    cancelTokenSource?: CancelTokenSource,
+    onProgress?: (progressEvent: ProgressEvent) => void
+  ): Promise<IUploadAttachmentResponse> => {
+    const req_message = new FormData();
+
+    req_message.append('media', file);
+    attachmentType && req_message.append('attachmentType', attachmentType);
+
+    const { data } = await axios.post(`/api/project/${projectId}/survey/${surveyId}/attachments/upload`, req_message, {
+      cancelToken: cancelTokenSource?.token,
+      onUploadProgress: onProgress
+    });
+
+    return data;
+  };
+
+  /**
+   * Upload survey reports.
+   *
+   * @param {number} projectId
+   * @param {number} surveyId
+   * @param {File} file
+   * @param {string} attachmentType
+   * @param {CancelTokenSource} [cancelTokenSource]
+   * @param {(progressEvent: ProgressEvent) => void} [onProgress]
+   * @return {*}  {Promise<string[]>}
+   */
+  const uploadSurveyReports = async (
```

Comment:
It wont let me comment on the right line, but we can make `attachmentMeta` required now, since it should always be present for reports.